### PR TITLE
Before date

### PR DESCRIPTION
**Pull request by saurabhketech**

### CODE REVIEW

File: src/controllers/imap.js [hunk 1]
```diff
@@ -4,124 +4,120 @@ import ImapProvider from "../providers/ImapProvider";
...
 
-/* Get Imap data */
-	getImap = (req, res) => {
-		this._db.Imap.findAll({ offset: (req.params.page - 1) * 10, limit: 10 })
-.then(res.json.bind(res))
-.catch(this.handleErrorResponse.bind(null, res));
-	}
+    /* Get Imap data */
+    getImap = (req, res) => {
+        this._db.Imap.findAll({ offset: (req.params.page - 1) * 10, limit: 10 })
```
Comment (mageexxcel) [inline src/controllers/imap.js:56]: i already told not to hard code anything many times

File: src/controllers/imap.js [hunk 2]
```diff
@@ -4,124 +4,120 @@ import ImapProvider from "../providers/ImapProvider";
...
+                        password: result.password,
+                        host: result.imap_server,
+                        port: result.server_port,
+                        tls: result.type
+                    });
+                    imap_connection(imap, (err) => {
+                        if (err) {
+                            res.json({ status: 0, message: "error", data: err });
+                        } else {
+                            this._db.Imap.update({ status: "FALSE" }, { where: { email: req.body.email } })
```
Comment (mageexxcel) [inline src/controllers/imap.js:77]: status: "FALSE"  wrong

File: src/controllers/imap.js [hunk 3]
```diff
@@ -4,124 +4,120 @@ import ImapProvider from "../providers/ImapProvider";
...
+                                .then((data) => {
+                                    if (data[0] == 1) {
+                                        res.json({ status: 1, message: "success", data: "successfully Active changed to true" });
+                                    } else if (data[0] == 0) {
```
Comment (mageexxcel) [inline src/controllers/imap.js:81]: controller code should not be more than 5-6 lines

File: src/controllers/imap.js [hunk 4]
```diff
@@ -4,124 +4,120 @@ import ImapProvider from "../providers/ImapProvider";
...
-		});
-	});
-	imap.once("error", function(err) {
-		callback(err);
-	});
-	imap.once("end", function() {
-		console.log("Connection ended");
-	});
-	imap.connect();
+    function openInbox(cb) {
```
Comment (mageexxcel) [inline src/controllers/imap.js:100]: what is this doing in controller?

File: src/controllers/smtp.js [hunk 5]
```diff
@@ -6,126 +6,119 @@ var _ = require("lodash");
...
-		}
-	});
-})
-            .catch(this.handleErrorResponse.bind(null, res));
-	}else{
-		res.json({status:0,messsage:"error" , data:"email id is not found in database"});
-	}
-})
+                var email = req.body.email;
+                var subject = "Smtp test";
```
Comment (mageexxcel) [inline src/controllers/smtp.js:71]: wrong code in controller

File: src/controllers/smtp.js [hunk 6]
```diff
@@ -6,126 +6,119 @@ var _ = require("lodash");
...
+    }
 
-})
-							.catch(this.handleErrorResponse.bind(null, res));
-	}
+    /* change smtp status*/
+    changeStatus = (req, res) => {
+        this._db.Smtp.findOne({ where: { email: req.params.email } })
+            .then((data) => {
+                if (data) {
```
Comment (mageexxcel) [inline src/controllers/smtp.js:93]: wrong code in controller

File: src/mongodb/db.js [hunk 7]
```diff
@@ -1,23 +1,35 @@
...
+        date: { type: String },
+        email_date: { type: String },
+        email_timestamp: { type: String },
+        subject: { type: String },
+        unread: { type: Boolean },
+        answered: { type: Boolean },
+        uid: { type: Number },
+        body: { type: String },
+        tags: { type: String },
+        Genuine_Applicant: { type: String },
```
Comment (mageexxcel) [inline src/mongodb/db.js:22]: Genuine_Applicant  lower case

File: src/service/cron.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+var inbox = require("../inbox");
+var CronJob = require("cron").CronJob;
+
+export class CronController {
+    cron(email) {
+        new CronJob("*/15 * * * *", function() {
+            inbox.fetch_email(email); // running this function every 15 min
+            inbox.Before_date_email(email, new Date());
```
Comment (mageexxcel) [inline src/service/cron.js:8]: Before_date_email camecase

File: src/controllers/imap.js (hunk 1 above)
Comment (saurabhketech) [reply src/controllers/imap.js:56]: change 10 to req.params.limit , get limit from front_end

File: src/controllers/imap.js (hunk 3 above)
Comment (saurabhketech) [reply src/controllers/imap.js:81]: done . put all controller code in class methods

File: src/controllers/imap.js (hunk 4 above)
Comment (saurabhketech) [reply src/controllers/imap.js:100]: put it in model

File: src/controllers/smtp.js (hunk 5 above)
Comment (saurabhketech) [reply src/controllers/smtp.js:71]: put it in constant

File: src/controllers/smtp.js (hunk 6 above)
Comment (saurabhketech) [reply src/controllers/smtp.js:93]: put all code in smtp models

File: src/mongodb/db.js (hunk 7 above)
Comment (saurabhketech) [reply src/mongodb/db.js:22]: set genuine_applicant to lower case

File: src/service/cron.js (hunk 8 above)
Comment (saurabhketech) [reply src/service/cron.js:8]: done

**Pull request merged**